### DISCUSSION TITLE
Build Project IOS

### DISCUSSION THREAD

**Discussion by NgVThangBz** [axmolengine/axmol#1372] [Q&A] [answered]
I need to build an ios project, and there are also .swift files, of course a header file is needed, but it seems that Cmake cannot add .swift files, so how can I save the settings of the IOS project? I will need to upload some files in the build folder?

**Comment by rh101** [comment, marked as answer]
> but it seems that Cmake cannot add .swift files, so how can I save the settings of the IOS project?

What makes you think that?

A quick online search for "cmake swift example", and you would see a few hundred thousand results showing you how this is done.

The simplest way is to change the `project` at the top of your root `CMakeLists.txt` to this:
```
if(APPLE)
    project(${APP_NAME} LANGUAGES C CXX OBJC OBJCXX Swift)
else()
    project(${APP_NAME} LANGUAGES C CXX)
endif()
```

Then you need to set several properties:

```
set_target_properties(${YourTarget} PROPERTIES
XCODE_ATTRIBUTE_SWIFT_VERSION "4.0"
XCODE_ATTRIBUTE_SWIFT_OBJC_INTERFACE_HEADER_NAME "somefilename-Swift.h"
XCODE_ATTRIBUTE_SWIFT_OBJC_BRIDGING_HEADER "${CMAKE_CURRENT_LIST_DIR}/path/to/ObjC-Swift-Bridge.h"
XCODE_ATTRIBUTE_ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES YES
)
```

You just add the swift files as you would any other file, for example:
```
list(APPEND GAME_SOURCE
     proj.ios_mac/ios/main.m
     proj.ios_mac/ios/AppController.mm
     proj.ios_mac/ios/RootViewController.mm
     proj.ios_mac/ios/BridgeHeader.swift
     proj.ios_mac/ios/MyGame-Bridging-Header.h
     proj.ios_mac/ios/Prefix.pch
     ${APP_UI_RES}
     )
```
